spec/requests/api/v3/user/userlock_resource_spec.rb

```diff
@@ -0,0 +1,122 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+require 'rack/test'
+
+describe 'API v3 UserLock resource', type: :request do
+  include Rack::Test::Methods
+
+  let(:current_user) { FactoryGirl.build_stubbed(:user) }
+  let(:user) { FactoryGirl.create(:user, status: User::STATUSES[:active]) }
+  let(:model) { ::API::V3::Users::UserModel.new(user) }
+  let(:representer) { ::API::V3::Users::UserRepresenter.new(model) }
+  let(:lock_path) { "/api/v3/users/#{user.id}/lock" }
+  subject(:response) { last_response }
+
+  describe '#post' do
+    before do
+      allow(User).to receive(:current).and_return current_user
+      post lock_path
+      # lock manually
+      user.lock
+    end
+
+    # Locking is only available for admins
+    context 'when logged in as admin' do
+      let(:current_user) { FactoryGirl.build_stubbed(:admin) }
+
+      context 'user account can be locked' do
+        it 'should respond with 200' do
+          expect(subject.status).to eq(200)
+        end
+
+        it 'should respond with an updated lock status in the user model' do
+          expect(parse_json(subject.body, 'status')).to eq 'locked'
+        end
+      end
+
+      context 'user account is incompatible' do
+        let(:user) { FactoryGirl.create(:user,
+          status: User::STATUSES[:registered]) }
+        it 'should fail for invalid transitions' do
+          expect(subject.status).to eq(400)
+        end
+      end
+    end
+
+    context 'requesting nonexistent user' do
+      let(:lock_path) { '/api/v3/users/9999/lock' }
+      it_behaves_like 'not found', 9999, 'User'
+    end
+
+    context 'non-admin user' do
+      it 'should respond with 403' do
+        expect(subject.status).to eq(403)
+      end
+    end
+  end
+
+  describe '#delete' do
+    before do
+      allow(User).to receive(:current).and_return current_user
+      delete lock_path
+      # unlock manually
+      user.activate
+    end
+
+    # Unlocking is only available for admins
+    context 'when logged in as admin' do
+      let(:current_user) { FactoryGirl.build_stubbed(:admin) }
+
+      context 'user account can be unlocked' do
+        it 'should respond with 200' do
+          expect(subject.status).to eq(200)
+        end
+
+        it 'should respond with an updated lock status in the user model' do
+          expect(parse_json(subject.body, 'status')).to eq 'active'
+        end
+      end
+
+      context 'user account is incompatible' do
+        let(:user) { FactoryGirl.create(:user,
+          status: User::STATUSES[:registered]) }
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Expression at 109, 48 should be on its own line.